ertviz/ertapi/ensemble/request_data.py

```diff
@@ -0,0 +1,123 @@
+from collections import namedtuple
+import ertviz.ertapi.ensemble
+import pandas as pd
+
+
+class ParametersDict:
+    def __init__(self, field, key, parent):
+        self._field = field
+        self._parent = parent
+        self._key = key
+        self._index = -1
+
+    def __iter__(self):
+        return self
+
+    def _get_gen_class(self):
+        return {
+            "parameters": ertviz.ertapi.ensemble.Parameter,
+            "observations": ertviz.ertapi.ensemble.Observation,
+            "realizations": ertviz.ertapi.ensemble.Realization,
+            "responses": ertviz.ertapi.ensemble.Response,
+            "parameter_realizations": ertviz.ertapi.ensemble.ParameterRealization,
+        }[self._field]
+
+    def __next__(self):
+        nodes = [node for node in self._parent.metadata[self._field]]
+        self._index += 1
+        if self._index < len(nodes):
+            GenClass = self._get_gen_class()
+            return GenClass(self._parent._request_handler, nodes[self._index])
+        raise StopIteration
+
+    def __getitem__(self, idx):
+        GenClass = self._get_gen_class()
+        for node in self._parent.metadata[self._field]:
+            if idx == node[self._key]:
+                return GenClass(self._parent._request_handler, node)
+        return None
+
+    def __getattr__(self, attr):
+        return [node[attr] for node in self._parent.metadata[self._field]]
+
+
+class RequestData:
+    def __init__(self, request_handler, metadata_dict=None):
+        self._metadata = metadata_dict
+        self._request_handler = request_handler
+        self._clear_data()
+        self.load_metadata()
+
+    def _clear_data(self):
+        self._realizations = {}
+        self._responses = {}
+        self._parameters = {}
+        self._observations = {}
+
+    def get_node_fields(self, field, key=None):
+        if field in self.metadata:
+            if key is None:
+                return self.metadata[field]
+            if not isinstance(key, list):
+                key = [key]
+            Record = namedtuple(field, key)
+            return Record(
+                *[[node[val] for node in self.metadata[field]] for val in key]
+            )
+
+        return None
+
+    @property
+    def parameters(self):
+        return ParametersDict("parameters", "key", self)
+
+    @property
+    def responses(self):
+        return ParametersDict("responses", "name", self)
```

Comment:
are all these parametersdict? odd naming when have a strong notion of what parameters are in ERT

Reply:
Agree, this should be change to something better. Essentially, it unifies access to objects:
```python
ensemble.responses.name #gets all names of responses, without creating responses that is
ensembles.responses["POLY_RES"] #gets a single response object
```